### DISCUSSION TITLE
Uses of the package

### DISCUSSION THREAD

**Discussion by sbfnk** [sbfnk/mfiidd.archive#15] [General]
It would be great to find out more about how people are using the package, either for self-study or for their own teaching. We would welcome anyone posting their experiences here.

**Comment by danielsadoc** [comment]
I use the package for teaching. Here is an exercise, based on example 5.5 in Dobrow's book on Stochastic Processes with R:

Study example 5.5 from the book, assuming that C is the length of the proposal interval, which in the book's case is assumed to be equal to 2 (ranging from -1 to +1).
[also refer to: http://sbfnk.github.io/mfiidd/slides/MCMC_movement.gif and compare it with example 5.5 from the book -- hint: you may also want to search for material on https://sbfnk.github.io/mfiidd/ and perhaps https://sbfnk.github.io/mfiidd/slides/closing.pdf]

Answer the following questions:

4.1) Let C be the length of the proposal distribution interval, which in the book's example is C=2. If C does not affect the acceptance function, what is its role in the algorithm? What happens if C is very small? And if C is very large?

4.2) Vary the value of C and reproduce figure 5.5 from the book. What happens if C is very large? And if C is very small? Try different values of C. You can easily do this by modifying the source code presented at the end of section 5.2 of the book.

4.3) Explain how the MCMC method works for sampling from a Gaussian variable: what is the role of the current state in the algorithm? What adaptation was made to the MCMC algorithm to enable it to sample from continuous distributions instead of discrete ones?

4.4) Explain with your own words the animation in http://sbfnk.github.io/mfiidd/slides/MCMC_movement.gif  -- what do the different colors mean?
